lib/utils/report.js

```diff
@@ -109,3 +109,23 @@ module.exports = function report(problem) {
 
 	result.warn(warningMessage, warningProperties);
 };
+
+/**
+ * @param {import('postcss').Node | undefined} node
+ * @param {number | undefined} index
+ * @param {number | undefined} endIndex
+ * @returns {number | undefined}
+ */
+function getStartLine(node, index, endIndex) {
+	if (node === undefined) return undefined;
+
+	// HACK: The `rangeBy` method has been added since PostCSS 8.4.0,
+	//       while the `node` may be produced by PostCSS 8.3.x or lower
+	//       that doesn't have the method.
+	//       See https://github.com/stylelint/stylelint/issues/5766
+	if (node.rangeBy === undefined) {
+		node.rangeBy = postcss.Node.prototype.rangeBy.bind(node);
+	}
```

Comment:
[note] The implementation of `rangeBy` is here:

https://github.com/postcss/postcss/blob/8.4.14/lib/node.js#L260

This access to `postcss.Node.prototype` may be flaky because of a dependency on the internal implementation of PostCSS.